File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/activities/SubmitReview.java
package com.moovie.backend.domain.usecase.activities;

import com.moovie.backend.domain.model.MovieReviewDraft;
import com.moovie.backend.domain.repository.UserActivitiesRepository;
import org.springframework.stereotype.Service;

@Service
public class SubmitReview {

    private final UserActivitiesRepository userActivitiesRepository;

    public SubmitReview(UserActivitiesRepository userActivitiesRepository) {
        this.userActivitiesRepository = userActivitiesRepository;
    }

    public void invoke(MovieReviewDraft draft) {
        userActivitiesRepository.submitReview(draft);
    }
}

File: src/main/kotlin/org/example/domain/usecase/activities/SubmitReview.kt
package org.example.domain.usecase.activities

import kotlinx.coroutines.runBlocking
import org.example.domain.model.MovieReviewDraft
import org.example.domain.repository.UserActivitiesRepository

class SubmitReview(private val repository: UserActivitiesRepository) {
    operator fun invoke(draft: MovieReviewDraft) = runBlocking {
        repository.submitReview(draft)
    }
}
